datasets/hlgd/hlgd.py

```diff
@@ -0,0 +1,126 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+HLGD is a binary classification dataset consisting of 20,056 labeled news headlines pairs indicating
+whether the two headlines describe the same underlying world event or not.
+"""
+
+import json
+import os
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{Laban2021NewsHG,
+  title={News Headline Grouping as a Challenging NLU Task},
+  author={Philippe Laban and Lucas Bandarkar},
+  booktitle={NAACL 2021},
+  publisher = {Association for Computational Linguistics},
+  year={2021}
+}
+"""
+
+_DESCRIPTION = """\
+HLGD is a binary classification dataset consisting of 20,056 labeled news headlines pairs indicating
+whether the two headlines describe the same underlying world event or not.
+"""
+
+_HOMEPAGE = "https://github.com/tingofurro/headline_grouping"
+_LICENSE = "Apache-2.0 License"
+_DOWNLOAD_URL = "https://github.com/tingofurro/headline_grouping/releases/download/0.1/hlgd_classification_0.1.zip"
+
+
+class HLGD(datasets.GeneratorBasedBuilder):
+    """Headline Grouping Dataset."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "timeline_id": datasets.features.ClassLabel(names=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]),
+                "headline_a": datasets.Value("string"),
+                "headline_b": datasets.Value("string"),
+                "date_a": datasets.Value("string"),
+                "date_b": datasets.Value("string"),
+                "url_a": datasets.Value("string"),
+                "url_b": datasets.Value("string"),
+                "label": datasets.features.ClassLabel(names=[0, 1]),
```

Comment:
Can you use more explicit label names here please ?

For example

```suggestion
                "label": datasets.features.ClassLabel(names=["different_event", "same_event"]),
```

You'll need to regenerate the dataset_infos.json file after this change
```suggestion
                "label": datasets.features.ClassLabel(names=[0, 1]),
```

Reply:
Good idea. Switched the label names and regenerated `dataset_infos.json`.